mlrun/serving/v2_serving.py

```diff
@@ -504,15 +504,43 @@ def push(self, start, request, resp=None, op=None, error=None, partition_key=Non
             self.output_stream.push([data], partition_key=partition_key)
             return
 
-        self._sample_iter = (self._sample_iter + 1) % self.stream_sample
-        if self.output_stream and self._sample_iter == 0:
+        if self.output_stream:
+            if not isinstance(request["inputs"][0], list):
```

Comment:
I think this is the safer way to do this, think on a situation that one of the features is a list and the rest are not. 
```
request["inputs"] = (
    request["inputs"]
    if not any(not isinstance(feat, list) for feat in request["inputs"])
    else [request["inputs"]]
) 
```